src/main/java/org/wildfly/security/auth/realm/jdbc/JdbcSecurityRealm.java

```diff
@@ -247,6 +248,14 @@ private Connection getConnection(QueryConfiguration configuration) {
         private <E> E executePrincipalQuery(QueryConfiguration configuration, ResultSetCallback<E> resultSetCallback) {
             String sql = configuration.getSql();
 
+            if (log.isDebugEnabled()) {
+                log.debug("Executing principalQuery...");
+            }
+
+            if (log.isTraceEnabled()) {
```

Comment:
Such conditions are useless, if generating of tracef/debug parameters is cheap operation. It only make sense if there is for example `String.join(", ", array)`, which is expensive operation which should be omitted when trace is not enabled.